OlinkAnalyze/R/bridgeable.R

```diff
@@ -0,0 +1,293 @@
+                  .data[["Gene"]],
+                  .data[["OlinkID"]],
+                  .data[["ids"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count"]]) |>
+    dplyr::mutate(Count_ht = .data[["Count"]]) |>
+    distinct()
+
+  data_mapped <- data_Explore384 |>
+    dplyr::inner_join(data_HT, by = c("SampleID", "ids"),
+                      relationship = "many-to-many") |>
+    na.omit() |>
+    dplyr::group_by(.data[["ids"]]) |>
+    dplyr::mutate(range_ht =
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.9) -
+                    quantile(.data[["NPX_ht"]][.data[["Count_ht"]] > 10],
+                             probs = 0.1),
+                  # using 10% to 90% quantiles for checking range /IQR
+                  range_3k = quantile(.data[["NPX_3k"]][.data[["Count_3k"]] >
+                                                          10],
+                                      probs = 0.9) -
+                    quantile(.data[["NPX_3k"]][.data[["Count_3k"]] > 10],
+                             probs = 0.1),
+                  range_diff = abs(.data[["range_ht"]] - .data[["range_3k"]]),
+                  iqr_ht = IQR(.data[["NPX_ht"]]),
+                  iqr_3k = IQR(.data[["NPX_3k"]]),
+                  median_count_3k =
+                    median(.data[["Count_3k"]][.data[["Count_3k"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  median_count_ht =
+                    median(.data[["Count_ht"]][.data[["Count_ht"]] > 10],
+                           na.rm = TRUE),
+                  # if count less than 10 in 1 sample that assay is dropped
+                  low_counts_flag = .data[["median_count_3k"]] < 150 |
+                    .data[["median_count_ht"]] < 150) |>
+    dplyr::ungroup() |>
+    dplyr::distinct()
+
+  # selecting only required columns
+  data_condensed <- data_mapped |>
+    dplyr::select(.data[["SampleID"]],
+                  .data[["Gene"]],
+                  .data[["UniProt"]],
+                  .data[["NPX_3k"]],
+                  .data[["NPX_ht"]],
+                  .data[["Count_ht"]],
+                  .data[["Count_3k"]],
+                  .data[["range_diff"]],
+                  .data[["low_counts_flag"]],
+                  .data[["ids"]]) |>
+    dplyr::ungroup() |>
+    dplyr::distinct() |>
+    r2_results()
+
+  not_bridgeable <- data_condensed |>
+    dplyr::filter(.data[["range_diff"]] >
+                    1, .data[["low_counts_flag"]] ==
+                    TRUE, .data[["R2_lm"]] < 0.8) |>
+    dplyr::pull(.data[["ids"]]) |>
+    unique() |>
+    as.data.frame() |>
+    dplyr::rename(ids = 1) |>
+    tidyr::extract(.data[["ids"]], into = c("OlinkID", "OlinkID_Explore384"),
+                   "(.*)_([^_]+)$", remove = TRUE)
```

Comment:
I am not sure why we need this long pipe. We can just extract the concatenated OlinkIDs and do a left join with the mapping file or, if we prefer splitting the OlinkIDs we can do select instead of pull and then mutate to get the HT and 3k OlinkIDs.

Reply:
Condensed code to use concatenated OlinkIDs for not_bridgeable filtering